src/Symfony/Component/Cache/Adapter/OpCacheAdapter.php

```diff
@@ -95,12 +105,8 @@ public function store(array $values)
         } else {
             $directory = dirname($this->file);
 
-            if (!file_exists($directory)) {
-                @mkdir($directory, 0777, true);
-
-                if (!file_exists($directory)) {
-                    throw new InvalidArgumentException(sprintf('Cache directory does not exist and cannot be created: %s', $directory));
-                }
+            if (!is_dir($directory) && !@mkdir($directory, 0777 & (~$this->umask), true)) {
```

Comment:
the previous version was more resistant to race conditions, the last file_exists check should be kept
also, no need for the () around the umask call